### PR TITLE
Suuport BertAdam

### PR DESCRIPTION
**Pull request by dding3**

### CODE REVIEW

**Comment by jason-dai** [comment]
How's it different from existing implementation of Adam https://github.com/intel-analytics/analytics-zoo/blob/master/zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/optimizers/Adam.scala? Can we give it a more meaningful name?

**Comment by dding3** [comment]
jekins pass. http://10.239.47.211:18888/job/ZOO-PR-Validation/2212/

**Comment by dding3** [comment]
The differences are
1. `BertAdam` implements weight decay fix,
2. `BertAdam` doesn't compensate for bias as in the regular Adam optimizer.
The last few steps are different, it might not easy to reuse code in `Adam`.

Can we rename it to `AdamWeightDecay`? I noticed in official Bert implementation it's named as `AdamWeightDecayOptimizer`.

**Comment by jason-dai** [comment]
`AdamWeightDecay` sounds good to me

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/optimizers/BERTAdam.scala [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+ */
+class BERTAdam[@specialized(Float, Double) T: ClassTag](
+   var learningRate: Double = 1e-3,
+   var warmupPortion: Double = -1,
+   var total: Int = -1,
+   var schedule: String = "linear",
+   var beta1: Double = 0.9,
+   var beta2: Double = 0.999,
+   var epsilon: Double = 1e-6,
+   var weightDecay: Double = 0.01)(implicit ev: TensorNumeric[T]) extends OptimMethod[T] {
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/optimizers/BERTAdam.scala]: should it extend SGD?

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/optimizers/BERTAdam.scala [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    this.beta1 = config.get[Double]("beta1").getOrElse(this.beta1)
+    this.beta2 = config.get[Double]("beta2").getOrElse(this.beta2)
+    this.epsilon = config.get[Double]("Epsilon").getOrElse(this.epsilon)
+    this.weightDecay = config.get[Double]("weightDecay")
+      .getOrElse(this.weightDecay)
+    this
+  }
+
+  override def clearHistory(): Unit = {
+    val state = SGDRef.getstate(this)
```
Comment (jason-dai) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/optimizers/BERTAdam.scala]: should it extend SGD? And shall we call `super.clearHistory`?

Comment (dding3) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/optimizers/BERTAdam.scala]: Updated

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/optimizers/BERTAdam.scala (hunk 1 above)
Comment (dding3) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/api/keras/optimizers/BERTAdam.scala]: Update

**Comment by jason-dai** [comment]
LGTM

**Comment by dding3** [comment]
Thanks for your comments.
jekins pass. http://10.239.47.211:18888/job/ZOO-PR-Validation/2213/
